### PR TITLE
Feature/a1 mtdc@127 tf single app insights

### PR DESCRIPTION
**Pull request by algattik**
Use a single shared instance of Application Insights for all data space deployments. This allows tracing requests across the different layers of the system when building a composite parts tree.

This required splitting the terraform deployment into successive "layers", where the output of one layer (deploying the shared Application Insights instance) is used as the input of another layer. To make this more manageable, I've extracted the terraform scripting into a shell script.

Split the "monolithic" `terraform` folder into two terraform folders, `terraform-common` (containing only App Insights for now, maybe other things later) and `terraform-dataspace-partition` (the previous per-dataspace terraform deployment, minus App Insights).

Moved the `environments/` config files from `tfvars` files to `sh` sourced files to avoid warnings about unused terraform variables. This is also more robust, since we are already sourcing these files in shell.

Also moved the `helm` folder to be under `cd` for consistency.

Deploy run: https://github.com/catenax/tractusx/actions/runs/1428537191

### CODE REVIEW

**Comment by marcgs** [comment]
Could you add a link to a running deploy with this new setup?

**Review by marcgs — APPROVED** [review]

File: .github/workflows/partsrelationshipservice-deploy.yml [hunk 1]
```diff
@@ -66,51 +66,61 @@ jobs:
...
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: coreservices/partsrelationshipservice/cd/terraform-common
+    steps:
+      - uses: actions/checkout@v2
+      - uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+      - name: "Run Terraform, deploying infrastructure to Azure"
```
Comment (cpeeyush) [inline .github/workflows/partsrelationshipservice-deploy.yml:84]: nit:  just to make it more clear for reader while looking at workflow run.
```suggestion
      - name: "Run Terraform, deploying common infrastructure to Azure"
```

File: coreservices/partsrelationshipservice/cd/terraform-common/main.tf [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# PRS common infrastructure
+####################################################################################################
+
+module "prs_application_insights" {
+  source = "./modules/application-insights"
+
+  name                = "${var.prefix}-${var.environment}-prs-appi"
+  resource_group_name = local.resource_group_name
+  location            = local.location
+}
```
Comment (cpeeyush) [inline coreservices/partsrelationshipservice/cd/terraform-common/main.tf:11]: nit: new line
```suggestion
}

```

File: coreservices/partsrelationshipservice/cd/terraform-common/outputs.tf [hunk 3]
```diff
@@ -0,0 +1,4 @@
+output "application_insights_name" {
+  value       = module.prs_application_insights.name
+  description = "The Application Insights resource name."
+}
```
Comment (cpeeyush) [inline coreservices/partsrelationshipservice/cd/terraform-common/outputs.tf:4]: same as above
```suggestion
}

```

**Review by cpeeyush — APPROVED** [review]

**Pull request merged**